Improve error handling during reading of TLS configuration.

briansmith opened the pull request:
Switch from panics to "proper" error handling. In particular, now
error messages like "file not found" include the path to the file for
which the I/O was attempted.

briansmith commented:
This builds on top of #73 and #74. Only the last (third) commit here is unique to this PR.

briansmith commented:
In particular, this requires Rustls 0.10, so #74 is required.

olix0r commented on src/server/sni.rs:
```diff
@@ -3,25 +3,29 @@ use rustls::{Certificate, ResolvesServerCert, SignatureScheme, sign};
+    };
     let sni = Sni {
-        default: default.as_ref().map(|c| ServerIdentity::load(c)),
+        default: default,
```
tioli: recently ruts changed to support the format
```rust
let sni = Sni {
    default,
    ...
```

briansmith commented:
OK, I fixed the "default: default" thing and rebased. It's good to see that rustc doesn't complain about mixing the abbreviated and non-abbreviated initializer syntax any more.
